src/common/tracer.cc

```diff
@@ -1,86 +1,62 @@
-      tracer = jaegertracing::Tracer::make( tracer_name, configuration,
-	    jaegertracing::logging::consoleLogger());
-      dout(3) << "tracer_jaeger init successful" << dendl;
-    }
-    //incase of stale tracer, configure with a new global tracer
-    if (opentracing::Tracer::Global() != tracer) {
-      opentracing::Tracer::InitGlobal(
-	  std::static_pointer_cast<opentracing::Tracer>(tracer));
-    }
-  }
+#include "common/ceph_context.h"
+#include "global/global_context.h"
+#include "common/tracer.h"
 
-  jspan new_span(const char* span_name) {
-    return opentracing::Tracer::Global()->StartSpan(span_name);
- }
+#ifdef HAVE_JAEGER
 
-  jspan child_span(const char* span_name, const jspan& parent_span) {
-    //no parent check if parent not found span will still be constructed
-    return opentracing::Tracer::Global()->StartSpan(span_name,
-	{opentracing::ChildOf(&parent_span->context())});
- }
+namespace jaeger_configuration {
 
-  void finish_span(const jspan& span) {
-    if (span) {
-      span->Finish();
-    }
+jaegertracing::samplers::Config const_sampler("const", 1, "", 0, jaegertracing::samplers::Config::defaultSamplingRefreshInterval());
+
+jaegertracing::reporters::Config reporter_default_config(jaegertracing::reporters::Config::kDefaultQueueSize, jaegertracing::reporters::Config::defaultBufferFlushInterval(), true, jaegertracing::reporters::Config::kDefaultLocalAgentHostPort, "");
+
+jaegertracing::propagation::HeadersConfig headers_config("", "", "", "");
+
+jaegertracing::baggage::RestrictionsConfig baggage_config(false, "", std::chrono::steady_clock::duration());
+
+jaegertracing::Config jaeger_rgw_config(false, const_sampler, reporter_default_config, headers_config, baggage_config, "rgw", std::vector<jaegertracing::Tag>());
+
+jaegertracing::Config jaeger_osd_config(false, const_sampler, reporter_default_config, headers_config, baggage_config, "osd", std::vector<jaegertracing::Tag>());
+
+}
+
+namespace tracing {
+
+const std::shared_ptr<opentracing::Tracer> Tracer::noop_tracer = opentracing::MakeNoopTracer();
+
+Tracer::Tracer(jaegertracing::Config& conf):open_tracer(jaegertracing::Tracer::make(conf)) {}
+
+Tracer::Tracer(opentracing::string_view service_name) {
+  using namespace jaeger_configuration;
+  jaegertracing::Config conf(false, const_sampler, reporter_default_config, headers_config, baggage_config, service_name, std::vector<jaegertracing::Tag>());
+  open_tracer = jaegertracing::Tracer::make(conf);
+}
+
+std::unique_ptr<opentracing::Span> Tracer::start_trace(opentracing::string_view trace_name) {
+  if (is_enabled()) {
+    return open_tracer->StartSpan(trace_name);
   }
+  return noop_tracer->StartSpan(trace_name);
+}
 
-  void set_span_tag(const jspan& span, const char* key, const char* value) {
-    if (span) {
-      span->SetTag(key, value);
+std::unique_ptr<opentracing::Span> Tracer::start_span(opentracing::string_view span_name, std::unique_ptr<opentracing::Span>& parent_span) {
```

Comment:
```suggestion
std::unique_ptr<opentracing::Span> Tracer::add_span(opentracing::string_view span_name, std::unique_ptr<opentracing::Span>& parent_span) {
```
how about using add_span, since we intend to add this span to the trace.

Reply:
done